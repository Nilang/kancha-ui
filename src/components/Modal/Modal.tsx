import * as React from 'react'
import Container from '../Container/Container'
import { ScrollView, StatusBar } from 'react-native'
import { withTheme } from '../../theming'
import Button from '../Button/Button'
import Icon from '../Icon/Icon'

interface ModalProps {
  /**
   * Enable scrollView
   */
  scrollEnabled?: boolean

  /**
   * Dismiss the modal view
   */
  dismiss?: () => void

  theme: any
}

const Modal: React.FC<ModalProps> = props => {
  const scrollViewContent = <ScrollView>{props.children}</ScrollView>

  return (
    <Container
      flex={1}
      marginTop={50}
      marginLeft={10}
      marginRight={10}
      backgroundColor={'#ffffff'}
      viewStyle={{
        shadowColor: '#000000',
        shadowRadius: 20,
        shadowOpacity: 0.2,
        borderTopLeftRadius: 10,
        borderTopRightRadius: 10,
      }}
    >
      <StatusBar hidden={true} animated={true} showHideTransition={'slide'} />
      {props.dismiss && (
        <Container viewStyle={{ position: 'absolute', right: 10, top: 10 }}>
          <Button
            iconButton={true}
            icon={
              <Icon icon={props.theme.icons.CLOSE} color={props.theme.colors.primary.accessory} size={40} />
            }
            onPress={props.dismiss}
          />
        </Container>
      )}
      {props.scrollEnabled ? scrollViewContent : props.children}
    </Container>
  )
}

export default withTheme(Modal)
