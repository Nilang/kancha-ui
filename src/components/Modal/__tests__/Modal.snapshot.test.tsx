import * as React from 'react'
import { render } from 'react-native-testing-library'
import Modal from '../Modal'
import Text from '../../Text/Text'

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon')

describe('Component(snapshots): Modal', () => {
  it('should render with no props', () => {
    const tree = render(<Modal />)

    expect(tree).toMatchSnapshot()
  })

  it('should render with props', () => {
    const dismiss = jest.fn()

    const tree = render(<Modal scrollEnabled={true} dismiss={dismiss} />)

    expect(tree).toMatchSnapshot()
  })

  it('should render with props and children', () => {
    const dismiss = jest.fn()

    const tree = render(
      <Modal scrollEnabled={true} dismiss={dismiss}>
        <Text>Test Modal</Text>
      </Modal>,
    )

    expect(tree).toMatchSnapshot()
  })
})
