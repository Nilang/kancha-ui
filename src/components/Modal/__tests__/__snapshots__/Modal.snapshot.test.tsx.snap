// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): Modal should render with no props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#ffffff",
      "borderRadius": undefined,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 50,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "shadowColor": "#000000",
      "shadowOpacity": 0.2,
      "shadowRadius": 20,
      "width": undefined,
    }
  }
/>
`;

exports[`Component(snapshots): Modal should render with props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#ffffff",
      "borderRadius": undefined,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 50,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "shadowColor": "#000000",
      "shadowOpacity": 0.2,
      "shadowRadius": 20,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "absolute",
        "right": 10,
        "top": 10,
        "width": undefined,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 8,
          "maxWidth": 300,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": "row",
            "height": undefined,
            "justifyContent": "center",
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": 10,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <Icon
            color="#BBBBBB"
            name="ios-close"
            size={40}
          />
        </View>
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": "center",
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</View>
`;

exports[`Component(snapshots): Modal should render with props and children 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#ffffff",
      "borderRadius": undefined,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 50,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "shadowColor": "#000000",
      "shadowOpacity": 0.2,
      "shadowRadius": 20,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "absolute",
        "right": 10,
        "top": 10,
        "width": undefined,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 8,
          "maxWidth": 300,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": "row",
            "height": undefined,
            "justifyContent": "center",
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": 10,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <Icon
            color="#BBBBBB"
            name="ios-close"
            size={40}
          />
        </View>
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": "center",
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <Text
        style={Object {}}
      >
        Test Modal
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
